sway/tree/workspace.c

```diff
@@ -870,8 +874,20 @@ void workspace_add_gaps(struct sway_workspace *ws) {
 		ws->current_gaps.right = 0;
 		ws->current_gaps.bottom = 0;
 		ws->current_gaps.left = 0;
+        for (int i = 0; i < ws->tiling->length; ++i) {
+            struct sway_container *con = ws->tiling->items[i];
+            con->corner_radius = 0;
+        }
 	} else {
 		ws->current_gaps = ws->gaps_outer;
+        for (int i = 0; i < ws->tiling->length; ++i) {
+            struct sway_container *con = ws->tiling->items[i];
+            con->corner_radius = config->corner_radius;
+        }
+        for (int i = 0; i < ws->floating->length; ++i) {
+            struct sway_container *con = ws->floating->items[i];
+            con->corner_radius = config->corner_radius;
+        }
```

Comment:
Is this needed? If so, why?

Reply:
When a window is opened in floating mode and is the first on the workspace, it loses its rounded corners without this.